BOLT 1: adds remote address to optional init_tlvs (IP discovery) (#917)
lightning/bolts: 01-messaging.md

    2. data:
        * [`...*chain_hash`:`chains`]

>     1. type: 2 (`remote_addr`)

Reviewer: If you're not adding a feature bit (and I don't believe we need to add one), you must use an odd number for compatibility (such as `3` for example)

    2. data:
        * [`...*chain_hash`:`chains`]

    1. type: 2 (`remote_addr`)
    2. data:
        * [`u8`:`addr_type`]

>         * `remote_addr_data`: `remote_addr`

Reviewer 2: Why not just specify it as the network address type from `node_announcement`?

  - SHOULD NOT set features greater than 13 in `globalfeatures`.
  - SHOULD use the minimum length required to represent the `features` field.
  - SHOULD set `networks` to all chains it will gossip or open channels for.

>   - SHOULD set `remote_addr` to reflect the IP address of an incoming

Reviewer 2: Maybe mention that it should *not* include any RFC1918 addresses.

    - MAY fail the connection.
  - if the feature vector does not set all known, transitive dependencies:
    - MUST fail the connection.

>   - MAY use the `remote_addr` to send an updated `node_annoucement` message
>     once it verified the validity of the address (i.e. by cross checking
>     with other peers or trying to connect to itself).

Reviewer: I would keep it even more concise and let implementation decide what additional validation they want to apply (if any):

```suggestion
  - MAY use the `remote_addr` to update its `node_annoucement`
```

Author: Yep, I also was thinking this was too much text :D
